Content/Presentations/Fortran/KokkosTutorial_Fortran.tex

```diff
@@ -0,0 +1,240 @@
+
+\input{KokkosTutorial_PreTitle}
+
+\title{Kokkos-Fortran-Interop}
+
+\input{KokkosTutorial_PostTitle}
+
+\usepackage{minted}
+
+\begin{document}
+
+\begin{frame}
+	\titlepage
+\end{frame}
+
+\begin{frame}{Installation}
+  \begin{itemize}
+    \item Repository: \url{https://github.com/kokkos/kokkos-fortran-interop}
+    \item Requirements:
+      \begin{itemize}
+        \item Kokkos 4.0 or newer
+        \item C++17/Fortran08 compiler suites
+      \end{itemize}
+    \item Configure\\
+      \texttt{cmake -DKokkos\_ROOT=/kokkos/path /interop/path}
+  \end{itemize}
+\end{frame}
+
+\begin{frame}{What is Kokkos-Fortran-Interop?}
+  \begin{itemize}
+    \item Kokkos-Fortran offers wrappers around:
+      \begin{itemize}
+        \item \texttt{Kokkos::initialize(argc, argv)}
+        \item \texttt{Kokkos::finalize()}
+        \item \texttt{Kokkos::print\_configuration(output)}
+        \item \texttt{Kokkos::View} 
+        \item \texttt{Kokkos::DualView}
+      \end{itemize}
+    \item User kernels are written in C\texttt{++} $\rightarrow$ need to use
+      \texttt{iso\_c\_binding}
+    \item Only a subset of Kokkos capabilities are exposed
+  \end{itemize}
+\end{frame}
+
+\begin{frame}{Starting a program}
+  \begin{itemize}
+    \item Similar to MPI, Kokkos need to be initialized using
+      \texttt{kokkos\_initialize()} and finalized using \texttt{kokkos\_finalize()}
+    \item The \texttt{kokkos\_initialize()} subroutine initializes Kokkos and
+      reads command line arguments
+    \item \texttt{kokkos\_print\_configure("output.txt")} prints the
+      Kokkos configuration to the file \texttt{output.txt}
+  \end{itemize}
+\end{frame}
+
+\begin{frame}[containsverbatim]{Simple Example}
+  \begin{minted}{fortran}
+program my_kokkos_code
+  use :: flcl_util_kokkos_mod
+
+  ! Initialize Kokkos
+  ! This subroutine reads command line arguments
+  call kokkos_initialize()
+
+  ! Print the configuration in a file
+  call kokkos_print_configure('kokkos.out')
+
+  ! Finalize Kokkos
+  call kokkos_finalize()
+end program my_kokkos_code
+  \end{minted}
+\end{frame}
+
+\begin{frame}{Kokkos::View}
+  \begin{itemize}
+    \item \texttt{Kokkos::View} is Kokkos equivalent to an array
+    \item \texttt{Kokkos::View} can have up to 8 dimensions in C++ but they are
```

Comment:
I think we only can do 6 dimensions in Kokkos 4 and only with Kokkos 5 using mdspan we supprt more. Maybe, I am wrong.

Reply:
The slide is correct
```
Views may have up to (at most) 8 dimensions and any number of these may be run-time or compile-time. 
```

Reply:
Ok, we can keep it as it is.